statistics/table.go

```diff
@@ -259,6 +259,10 @@ func (n *neededColumnMap) Delete(col tableColumnID) {
 	n.m.Unlock()
 }
 
+func (n *neededColumnMap) Length() int {
+	return len(n.cols)
```

Comment:
I think we need an `RLock()` around this.

Reply:
Yes you are right. For some languages it's OK but for golang it's data race. https://stackoverflow.com/questions/52710882/is-len-thread-safe-in-golang
Change it to RWMutex BTW.